component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -336,6 +327,136 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
     }
 
+    /**
+     * Process valid user token.
+     *
+     * @param context           AuthenticationContext.
+     * @param authenticatedUser AuthenticatedUser.
+     */
+    private void processValidUserToken(AuthenticationContext context, AuthenticatedUser authenticatedUser) {
+
+        context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, StringUtils.EMPTY);
+        context.setProperty(EmailOTPAuthenticatorConstants.EMAILOTP_ACCESS_TOKEN, StringUtils.EMPTY);
+        context.setSubject(authenticatedUser);
+        String emailFromProfile = context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL).toString();
+        if (StringUtils.isNotEmpty(emailFromProfile)) {
+            context.setSubject(
+                    AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(emailFromProfile));
+        }
+    }
+
+    /**
+     * Checks the backup codes for email otp.
+     *
+     * @param context           AuthenticationContext.
+     * @param userToken         EmailOTP token.
+     * @param authenticatedUser AuthenticatedUser
+     * @throws AuthenticationFailedException
+     */
+    private void checkWithBackUpCodes(AuthenticationContext context, String userToken,
+                                      AuthenticatedUser authenticatedUser) throws AuthenticationFailedException {
+
+        String[] savedOTPs = null;
+        String username = authenticatedUser.getUserName();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        UserRealm userRealm = getUserRealm(username);
+        try {
+            if (userRealm != null) {
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+                if (userStoreManager != null) {
```

Comment:
try to avoid nested-ifs. check userStoreManager == null in the beginning and return from there